control-plane-operator/controllers/hostedcontrolplane/snapshotcontroller/assets/07_deployment-hypershift.yaml

```diff
@@ -0,0 +1,62 @@
+# *** AUTOMATICALLY GENERATED FILE - DO NOT EDIT ***
+apiVersion: apps/v1
```

Comment:
we set some affinity/antiaffinity opinions for all components running on management side which would be missing in this yamls https://github.com/openshift/hypershift/blob/950ea390af0981824a2e953a537190666208ec09/support/config/deployment.go#L280
https://hypershift-docs.netlify.app/how-to/distribute-hosted-cluster-workloads/

Reply:
I think that's what `DeploymentConfig.ApplyTo` in [ReconcileOperatorDeployment](https://github.com/openshift/hypershift/pull/1698/files#diff-dd6fadc7d881adb3df754b570aeec6532cd7ae418771da5700fc13a75f2a399dR46) does.

Reply:
actually is not, you need to setDefaults() before `DeploymentConfig.ApplyTo`, usually we do
```
		deploymentConfig := config.DeploymentConfig{}
		deploymentConfig.Scheduling.PriorityClass = config.DefaultPriorityClass
		deploymentConfig.SetRestartAnnotation(hcp.ObjectMeta)
		deploymentConfig.SetDefaults(hcp, ignitionServerLabels, nil)
		deploymentConfig.ApplyTo(ignitionServerDeployment)
```

Where values dependes on you use case.
This is pending refactoring to force components to use it.

Reply:
That is done in `NewParams`: https://github.com/openshift/hypershift/pull/1698/files#diff-eef37071a0c64f539dca5d6d879376ed0be4086cdf7819fe764be284a1f963b2R43